src/main/java/io/sniffy/Sniffer.java

```diff
@@ -55,6 +56,19 @@ private static synchronized void notifyListeners(StatementMetaData statementMeta
         }
     }
 
+    private static synchronized void notifyListeners(String address, SocketStats socketStats) {
+        Iterator<WeakReference<Spy>> iterator = registeredSpies.iterator();
+        while (iterator.hasNext()) {
+            WeakReference<Spy> spyReference = iterator.next();
+            Spy spy = spyReference.get();
+            if (null == spy) {
+                iterator.remove();
+            } else {
+                spy.addExecutedStatement(address, new SocketStats(socketStats));
+            }
+        }
+    }
+
     // query counters
 
     protected static final Counter COUNTER = new Counter();
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/sniffy/sniffy/file/2261027752/issues/source?bid=3005960&fileBranchId=3034483#l74)